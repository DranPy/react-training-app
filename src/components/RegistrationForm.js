import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';

class RegistrationForm extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
  };

  render() {
    const { handleSubmit } = this.props;

    return (
      <div className="container">
        <form onSubmit={handleSubmit}>
          <div className="form-group row">
            <div className="col-4 col-form-label">
              <label className="form-control-plaintext">Email</label>
            </div>
            <div className="col-8">
              <Field className="form-control" name="email" component="input" type="email" />
            </div>
            <div className="col-4 col-form-label">
              <label className="form-control-plaintext">Password</label>
            </div>
            <div className="col-8">
              <Field className="form-control" name="password" component="input" type="password" />
            </div>
            <div className="col-4 col-form-label">
              <label className="form-control-plaintext">Confirm password</label>
            </div>
            <div className="col-8">
              <Field
                className="form-control"
                name="confirmPassword"
                component="input"
                type="password"
              />
            </div>
          </div>
          <div>
            <input type="submit" className="btn btn-sm" value="Register" />
          </div>
        </form>
      </div>
    );
  }
}

export default reduxForm({
  form: 'registrationForm',
  onSubmit: (values, dispatch, props) => props.onRegister(values),
})(RegistrationForm);
