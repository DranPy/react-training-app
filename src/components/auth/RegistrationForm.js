import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import InputWithLabel from '../../components/forms/fields/InputWithLabel';

class RegistrationForm extends Component {
  static propTypes = {
    handleSubmit: PropTypes.func,
    valid: PropTypes.bool,
    submitting: PropTypes.bool,
    errorMessage: PropTypes.string,
    isLoading: PropTypes.bool,
    isAuthorized: PropTypes.bool,
  };

  errorMessage = () => {
    const { errorMessage } = this.props;
    if (errorMessage) {
      return <div className="alert alert-warning">{errorMessage}</div>;
    }
  };

  // TODO: create page and move redirect to page
  redirectAuthorizedUser = () => {
    const { history, isAuthorized } = this.props;
    if (isAuthorized) {
      if (history.length > 1) {
        //   history.goBack();
        // } else {
        history.push('/');
      }
    }
  };

  render() {
    this.redirectAuthorizedUser();
    const { handleSubmit, valid, submitting } = this.props;

    return (
      <div className="container" style={{ width: 600 }}>
        <form onSubmit={handleSubmit}>
          <Field
            label="Email"
            className="form-control"
            name="email"
            component={InputWithLabel}
            type="email"
          />
          <Field
            label="Password"
            className="form-control"
            name="password"
            component={InputWithLabel}
            type="password"
          />
          <Field
            label="Confirm password"
            className="form-control"
            name="confirmPassword"
            component={InputWithLabel}
            type="password"
          />
          <div>
            <input
              type="submit"
              className="btn btn-block btn-primary"
              disabled={submitting || !valid}
              value="Register"
            />
          </div>
          {this.errorMessage()}
        </form>
      </div>
    );
  }
}

export default RegistrationForm;
