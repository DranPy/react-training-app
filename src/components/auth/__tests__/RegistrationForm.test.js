import React from 'react';
import { shallow } from 'enzyme';

import RegistrationForm from '../RegistrationForm';

describe('<RegistrationForm />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<RegistrationForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should submit form', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<RegistrationForm handleSubmit={handleSubmit} />);
    wrapper.find('form').simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
