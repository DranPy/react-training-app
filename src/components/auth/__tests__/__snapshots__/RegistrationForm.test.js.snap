// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm /> should match snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "width": 600,
    }
  }
>
  <form>
    <Field
      className="form-control"
      component={[Function]}
      label="Email"
      name="email"
      type="email"
    />
    <Field
      className="form-control"
      component={[Function]}
      label="Password"
      name="password"
      type="password"
    />
    <Field
      className="form-control"
      component={[Function]}
      label="Confirm password"
      name="confirmPassword"
      type="password"
    />
    <div>
      <input
        className="btn btn-block btn-primary"
        disabled={true}
        type="submit"
        value="Register"
      />
    </div>
  </form>
</div>
`;
